jenkins/docker/Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+    options {
+        timeout(time: 3, unit: 'HOURS') 
+    }
+    agent none
+    stages {
+        stage('parameters') {
+            steps {
+                script {
+                    properties([
+                            parameters([
+                                    string(
+                                            defaultValue: 'https://github.com/opensearch-project/opensearch-build',
+                                            name: 'DOCKER_BUILD_GIT_REPOSITORY',
+                                            description: 'The git repository name that contains dockerfiles and the docker build script',
+                                            trim: true
+                                    ),
+                                    string(
+                                            defaultValue: 'main',
+                                            name: 'DOCKER_BUILD_GIT_REPOSITORY_REFERENCE',
+                                            description: 'The git reference (branch/tag/commit_id) of above repository',
+                                            trim: true
+                                    ),
+                                    string(
+                                            defaultValue: 'bash docker/ci/build-image-multi-arch.sh -v <TAG_NAME> -f <DOCKERFILE PATH>',
+                                            name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS',
+                                            description: 'The script path of the docker build script, assuming you are already in root dir of DOCKER_BUILD_GIT_REPOSITORY',
+                                            trim: true
+                                    ),
+                                    booleanParam(
+                                            defaultValue: true,
+                                            name: 'IS_STAGING',
+                                            description: 'Are we pushing docker images to staging docker hub or production'
+                                    )
+                            ])
+                    ])
+                }
+            }
+        }
+        stage('docker-build') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0'
+                    args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    git url: "$DOCKER_BUILD_GIT_REPOSITORY", branch: "$DOCKER_BUILD_GIT_REPOSITORY_REFERENCE"
+                    if (env.IS_STAGING == "true"){
+                            withCredentials([usernamePassword(credentialsId: 'jenkins-staging-docker-staging-credential', usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
```

Comment:
Move the credential grab/script outside of the if statement and set variables for the updated values, removes duplication

Reply:
Yeah, that is an old if statement when I was testing, should make it better than this.

Reply:
fixed.